src/quickpicks/uris.ts

```diff
@@ -95,3 +95,32 @@ export async function uriQuickpick(
     return uri ? uri[0] : undefined;
   }
 }
+
+/** Representation of content retrieved from a file or editor. `openDocument` will be provided if
+ * the content came from an open editor, or if the associated file is open in an editor for the
+ * current workspace. */
+export interface LoadedDocumentContent {
+  content: string;
+  openDocument?: TextDocument;
+}
+
+/** Load the content of a document, either from an open editor or directly from the file system. */
+export async function loadDocumentContent(uri: Uri): Promise<LoadedDocumentContent> {
+  let content: string | undefined;
+  let document: TextDocument | undefined;
+  if (uri.scheme === "file") {
```

Comment:
I think it'd be better to loop through the open editors first, and if one corresponds to the Uri, then take that editor's buffer's current contents (since may have unsaved changes) as opposed to the on-disk contents.

Or maybe not. We should do a poll, and go with what people think is least-wrong behavior.

If the poll shows should warrant a change, can come in followup.